Dart/src/com/jetbrains/lang/dart/ide/marker/DartServerImplementationsMarkerProvider.java

```diff
@@ -0,0 +1,176 @@
+                                    final boolean addItem) {
+    if (!visited.add(item)) {
+      return;
+    }
+    if (addItem) {
+      final Element element = item.getClassElement();
+      final Location location = element.getLocation();
+      final DartComponent component = DartServerOverrideMarkerProvider.findDartComponent(project, location);
+      if (component != null) {
+        components.add(component);
+      }
+    }
+    for (int subIndex : item.getSubclasses()) {
+      final TypeHierarchyItem subItem = items.get(subIndex);
+      addSubClasses(project, visited, items, components, subItem, true);
+    }
+  }
+
+  private static void addSubMembers(@NotNull final Project project,
+                                    @NotNull final Set<TypeHierarchyItem> visited,
+                                    @NotNull final List<TypeHierarchyItem> items,
+                                    @NotNull final List<DartComponent> components,
+                                    @NotNull final TypeHierarchyItem item,
+                                    final boolean addItem) {
+    if (!visited.add(item)) {
+      return;
+    }
+    if (addItem) {
+      final Element element = item.getMemberElement();
+      if (element != null) {
+        final Location location = element.getLocation();
+        final DartComponent component = DartServerOverrideMarkerProvider.findDartComponent(project, location);
+        if (component != null) {
+          components.add(component);
+        }
+      }
+    }
+    for (int subIndex : item.getSubclasses()) {
+      final TypeHierarchyItem subItem = items.get(subIndex);
+      addSubMembers(project, visited, items, components, subItem, true);
+    }
+  }
+
+  @Nullable
+  private static LineMarkerInfo createMarker(@NotNull final DartComponentName name) {
+    final DartAnalysisServerService service = DartAnalysisServerService.getInstance();
+    final VirtualFile file = name.getContainingFile().getVirtualFile();
+    final int nameOffset = name.getTextOffset();
+    final int nameLength = name.getTextLength();
+    // classes
+    for (ImplementedClass clazz : service.getImplementedClasses(file)) {
+      if (clazz.getOffset() == nameOffset && clazz.getLength() == nameLength) {
+        return createMarkerClass(name);
+      }
+    }
+    // members
+    for (ImplementedMember member : service.getImplementedMembers(file)) {
+      if (member.getOffset() == nameOffset && member.getLength() == nameLength) {
+        return createMarkerMember(name);
+      }
+    }
+    // not found
+    return null;
+  }
+
+  @NotNull
+  private static LineMarkerInfo createMarkerClass(@NotNull final DartComponentName name) {
+    final VirtualFile file = name.getContainingFile().getVirtualFile();
+    final int nameOffset = name.getTextOffset();
+    return new LineMarkerInfo<PsiElement>(name, name.getTextRange(), AllIcons.Gutter.ImplementedMethod, Pass.UPDATE_OVERRIDEN_MARKERS,
```

Comment:
`Pass.UPDATE_OVERRIDEN_MARKERS` is used for markers created in `collectSlowLineMarkers()`. For markers from `getLineMarkerInfo()` use `Pass.UPDATE_OVERRIDEN_MARKERS`. Otherwise you'll see gutter width blinking and duplicate markers.
This is hard to guess, Platform API is bad here.